src/lib/ibcclient.ts

```diff
@@ -171,35 +179,32 @@ function createBroadcastTxErrorMessage(result: BroadcastTxFailure): string {
   return `Error when broadcasting tx ${result.transactionHash} at height ${result.height}. Code: ${result.code}; Raw log: ${result.rawLog}`;
 }
 
-// TODO: replace this with buildFees in IbcClient constructor to take custom gasPrice, override gas needed
-const fees = {
-  initClient: {
-    amount: coins(2500, 'ucosm'),
-    gas: '100000',
-  },
-  updateClient: {
-    amount: coins(10000, 'ucosm'),
-    gas: '400000',
-  },
-  initConnection: {
-    amount: coins(2500, 'ucosm'),
-    gas: '100000',
-  },
-  connectionHandshake: {
-    amount: coins(5000, 'ucosm'),
-    gas: '200000',
-  },
-  initChannel: {
-    amount: coins(2500, 'ucosm'),
-    gas: '100000',
-  },
-  channelHandshake: {
-    amount: coins(5000, 'ucosm'),
-    gas: '200000',
-  },
+export interface IbcFeeTable extends FeeTable {
+  readonly initClient: StdFee;
+  readonly updateClient: StdFee;
+  readonly initConnection: StdFee;
+  readonly connectionHandshake: StdFee;
+  readonly initChannel: StdFee;
+  readonly channelHandshake: StdFee;
+}
+
+export type IbcClientOptions = SigningStargateClientOptions & {
+  gasLimits?: Partial<GasLimits<IbcFeeTable>>;
+};
+
+const defaultGasPrice = GasPrice.fromString('0.025ucosm');
+const defaultGasLimits: GasLimits<IbcFeeTable & CosmosFeeTable> = {
+  send: 80000,
```

Comment:
Isn't `send` handled inside the SigningCosmosClient with it's defaults?

I bet you could reduce this to 10000 and it would not have any effect. If so, then remove it as misleading. (You can override it, but that default is not used.

Reply:
Ah yes, I should make this a `Partial<GasLimits<IbcFeeTable & CosmosFeeTable>>`